### PR TITLE
tagger: handle GPU tags

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR adds handling of GPU entities to the tagger.

### Motivation

Unify the tags for GPU devices
https://datadoghq.atlassian.net/browse/EBPF-599

### Describe how you validated your changes

Unit tests included in the changes. E2E tests will be included in https://github.com/DataDog/datadog-agent/pull/32906 and https://github.com/DataDog/datadog-agent/pull/32109 when the corresponding checks use this new feature.

### Additional Notes

Related to https://github.com/DataDog/datadog-agent/pull/32019

### CODE REVIEW

**Comment by adel121** [comment]
Can you please also update the [README.md](https://github.com/DataDog/datadog-agent/tree/main/comp/core/tagger#entity-ids) by adding the gpu entity id to the table of entity ids.

**Comment by adel121** [comment]
I don't think unit tests are enough to add `qa/done` label

Could you please do one of the following:
- Add e2e for this change, and keep the qa/done label.
- Test the change manually by running a custom build off your branch and verify the expected behaviour, then put the manual qa steps in the PR description, then remove the qa/done label so we can test this manually once it is merged on main during the release cycle. In this case, please also create a jira ticket to add E2E tests for this feature.

**Review by adel121 — CHANGES_REQUESTED** [review]
generally LGTM, but left 2 comments.

**Comment by gjulianm** [comment]
Added the entry to the README.md

I'll add e2e tests, but in that case I need to wait for https://github.com/DataDog/datadog-agent/pull/32109 to get merged, and change how we add tags in the GPU check.

**Comment by adel121** [comment]
> but in that case I need to wait for https://github.com/DataDog/datadog-agent/pull/32109 to get merged

If I understand correctly, as long as [wlm PR](https://github.com/DataDog/datadog-agent/pull/32109) is not merged, merging the current PR has no functional change. 

If my understanding is correct, E2E tests should be part of [the other PR
](https://github.com/DataDog/datadog-agent/pull/32109). You can remove the `qa/done` label from this PR, and link [the other PR](https://github.com/DataDog/datadog-agent/pull/32109) in the notes section saying that e2e tests will be added to wlm PR and QA can be marked directly as done just by verifying that E2E have already been put in place. 

WDYT?

**Comment by gjulianm** [comment]
Makes sense. We could merge this PR before the other then? The only thing is that the e2e tests would be split in two PRs: 
- I could add e2e tests in [#32109](https://github.com/DataDog/datadog-agent/pull/32109) for the presence of GPU entities in WMS 
- I can make another PR that uses the tagger to add tags to GPU metrics, and change the e2e test we have to check for presence of the proper tags.

**Comment by adel121** [comment]
> I could add e2e tests in https://github.com/DataDog/datadog-agent/pull/32109 for the presence of GPU entities in WMS

Presence of GPU entities in WMS is an internal implementation detail. E2E tests should not test internal implementation details, they should test E2E functionality considering the agent as a black box. In our case, what E2E tests should assert should be something like this: Run the agent on a node with nvidia gpu, and assert that the GPU metrics are properly tagged with GPU tags. This is what the end user will see, and hence it is what should be tested in E2E. 

So bottom line is we only need to add E2E once to cover both PRs. 

Internal details can be covered with unit tests.

**Comment by gjulianm** [comment]
> > I could add e2e tests in #32109 for the presence of GPU entities in WMS
> 
> Presence of GPU entities in WMS is an internal implementation detail. E2E tests should not test internal implementation details, they should test E2E functionality considering the agent as a black box. In our case, what E2E tests should assert should be something like this: Run the agent on a node with nvidia gpu, and assert that the GPU metrics are properly tagged with GPU tags. This is what the end user will see, and hence it is what should be tested in E2E.
> 
> So bottom line is we only need to add E2E once to cover both PRs.
> 
> Internal details can be covered with unit tests.

In that case I can add the E2E tests to the [PR changing the GPU check to use tags](https://github.com/DataDog/datadog-agent/pull/32906), and marking this PR as `qa/done` once that is merged.

**Comment by adel121** [comment]
Sounds good

Regarding the README.md, I think you missed pushing the change. Could you please push it so I can approve?

**Comment by gjulianm** [comment]
Pushed, I did forget to push, sorry :D

**Review by adel121 — APPROVED** [review]
LGTM
